### PR TITLE
Updates to hardware requirements

### PR DESCRIPTION
**Pull request by jasonblais**

### CODE REVIEW

File: source/deployment/scaling.rst [hunk 1]
```diff
@@ -4,32 +4,34 @@ Scaling for Enterprise
...
 Mattermost is designed to scale from small teams hosted on a single server to large enterprises running in cluster-based, highly available deployment configurations. 
 
-- Server requirements vary based on usage and it is highly recommended that pilots are run before enterprise-wide deployments in order to estimate full scale usage based on your specific organizational needs. 
+Server requirements vary based on usage and it is highly recommended that pilots are run before enterprise-wide deployments in order to estimate full scale usage based on your specific organizational needs. 
 
 Single Machine Deployment 
 ^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Organizations less than 500 users can typically run on a single server. 
+Organizations less than 2,000 users can typically run on a single server.
```
Comment (jasonblais) [inline source/deployment/scaling.rst]: Is the number "2000" correct?

File: source/install/requirements.rst [hunk 2]
```diff
@@ -101,60 +101,31 @@ Hardware Sizing for Team Deployments
 
 Most small to medium Mattermost team deployments can be supported on a single server with the following specifications based on registered users:
 
--  250-500 users - 2 vCPUs/cores, 4 GB RAM, and 45-90 GB storage
--  500-1,000 users - 4 vCPUs/cores, 8 GB RAM, and 90-180 GB storage
--  1,000-2,000 users - 4-8 vCPUs/cores, 16-32 GB RAM, and 180-360 GB storage
+-  1 - 1,000 users - 1 vCPU/cores, 2 GB RAM, and 45-90 GB storage
```
Comment (jasonblais) [inline source/install/requirements.rst]: I'm not sure if the storage recommendations are relevant, but we have a note `Storage recommendation is based on storing 3 years of archives with moderate file sharing.`.

Wondering if we should instead ask the admin to estimate file storage requirements based on file sharing within their team. I feel it's highly dependent on the use cases (e.g. a design firm probably requires a lot more file storage capacity compared to a dev ops team)

File: source/install/requirements.rst [hunk 3]
```diff
@@ -101,60 +101,31 @@ Hardware Sizing for Team Deployments
...
 
-.. _hardware-sizing-for-enterprise:
-
-Mattermost Load Test Framework
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-For Mattermost Enterprise Edition deployments, an open source load testing framework is available to simulate usage: https://github.com/mattermost/mattermost-load-test
+  1. Memory requirements are largely driven by peak file sharing activity. Recommendation is based on default 50 MB max file size, which can be adjusted from the System Console. Changing this number may change memory requirements. 
+  2. Storage recommendation is based on storing 3 years of archives with moderate file sharing.
+  3. Solid state drives (SSD) can be used in place of disk storage for higher concurrency.
```
Comment (jasonblais) [inline source/install/requirements.rst]: Is point 3 relevant?

File: source/install/requirements.rst [hunk 4]
```diff
@@ -101,60 +101,31 @@ Hardware Sizing for Team Deployments
...
-- One database server with 8-16 vCPUs/cores, 16-32 GB memory.
-- Minimum 100 GB SSD (solid state drive) storage for the binary and related files.
-- (Recommended) Add one identical database server to setup a Master-Slave configuration where the master can failover to slave with minimal disruption to service.
+Mattermost's `performance monitoring <https://docs.mattermost.com/deployment/metrics.html>`_ tools can be used for detailed performance measurements and inspect the running system to ensure sizing and installation is correct. 
 
-**Notes:**
+.. note::
 
-- Regular hard drives can be used in place of solid-state hard drives if having top performance is not a priority. If using a mix of HDD and SSD drives, the greatest performance gain would come from using SSD in the database server.
+  Regular hard drives (HDD) can be used in place of solid-state hard drives (SSD) if having top performance is not a priority. If using a mix of HDD and SSD drives, the greatest performance gain comes from using SSD in the database server.
```
Comment (jasonblais) [inline source/install/requirements.rst]: Is this note a duplication of https://github.com/mattermost/docs/pull/2302/files#diff-12b8342499d6b7515730e432a66bcda9R113

File: source/deployment/scaling.rst (hunk 1 above)
Comment (jwilander) [reply source/deployment/scaling.rst]: It really depends on usage. Pre-release is a 4,500 user server and I would be very comfortable running it on a single 2 vCPU 4GB RAM machine.

File: source/install/requirements.rst (hunk 2 above)
Comment (jwilander) [reply source/install/requirements.rst]: 0/5 the storage will also only matter if they're not using minio or S3

File: source/install/requirements.rst (hunk 3 above)
Comment (jwilander) [reply source/install/requirements.rst]: I think we can omit it

File: source/install/requirements.rst (hunk 4 above)
Comment (jwilander) [reply source/install/requirements.rst]: Essentially

**Comment by jwilander** [comment]
I'd suggest we also add a note saying that these recommendations are very much ballpark figures and will vary based on usage. Something like: "These recommendations are based off traditional Team Edition deployments and may grow or shrink depending on how active your users are." I'm essentially basing my recommendations off pre-release which is probably typical of a community TE deployment. Someone deploying TE to a company may see higher usage and need a higher specced machine as a result

**Comment by jasonblais** [comment]
Updated based on feedback:

1. Reworded the note about a single machine for less than 2,000 users to convey you can run higher numbers based on usage and server configuration https://github.com/mattermost/docs/pull/2302/files#diff-c075e9d1123114d517e20a9fc92d1b62R12
2. Updated intro of hardware requirements section, and moved notes relevant to both TE and EE to this section
3. Removed storage recommendations, given we have a "Alternate Storage Calculator" at the bottom of this page
4. Removed note about SSDs
5. Made it more clear these recommendations highly depend on usage

File: source/deployment/scaling.rst [hunk 5]
```diff
@@ -4,32 +4,34 @@ Scaling for Enterprise
...
 Mattermost is designed to scale from small teams hosted on a single server to large enterprises running in cluster-based, highly available deployment configurations. 
 
-- Server requirements vary based on usage and it is highly recommended that pilots are run before enterprise-wide deployments in order to estimate full scale usage based on your specific organizational needs. 
+Server requirements vary based on usage and it is highly recommended that pilots are run before enterprise-wide deployments in order to estimate full scale usage based on your specific organizational needs. 
 
 Single Machine Deployment 
 ^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Organizations less than 500 users can typically run on a single server. 
+Organizations can typically run Mattermost on a single server with up to 2,000 users, though higher numbers have been observed based on different usage and server configurations.
```
Comment (jwilander) [inline source/deployment/scaling.rst]: 0/5 but maybe `higher numbers` to `more users` just to be a bit more clear

**Pull request merged**